src/test/java/org/primeframework/mvc/test/RequestBuilder.java

```diff
@@ -832,18 +833,30 @@ HTTPResponseWrapper run() {
 
     // Now that the cookies are ready, if the CSRF token is enabled and the parameter isn't set, we set it to be consistent
     // since the [@control.form] would normally set that into the form and into the request.
-    MVCConfiguration configuration = injector.getInstance(MVCConfiguration.class);
-    if (configuration.csrfEnabled()) {
-      CSRFProvider csrfProvider = injector.getInstance(CSRFProvider.class);
-      if (csrfProvider.getTokenFromRequest(request) == null) {
-        String token = csrfProvider.getToken(request);
-        if (token != null) {
-          String parameterName = csrfProvider.getParameterName();
-          request.addURLParameter(parameterName, token);
+    if (request.getMethod() == HTTPMethod.POST) {
+      MVCConfiguration configuration = injector.getInstance(MVCConfiguration.class);
+      if (configuration.csrfEnabled()) {
+        CSRFProvider csrfProvider = injector.getInstance(CSRFProvider.class);
+        if (csrfProvider.getTokenFromRequest(request) == null) {
+          String token = csrfProvider.getToken(request);
+          if (token != null) {
+            String parameterName = csrfProvider.getParameterName();
+            requestBodyParameters.put(parameterName, List.of(token));
+          }
         }
       }
     }
 
+    // Optionally allow the caller to consume the HTTP request in order to mutate it and what not.
+    try {
```

Comment:
This is a little non-standard in that because this is an optional binding I am just catching the exception.
Anyone have any concerns with this pattern?

Reply:
Since this gets called a lot (exception/flow control), wouldn't it be more performant to have either:

1. A default consumer implementation that does nothing that people can override?
2. Maybe `injector.getExistingBinding` might be more performant?
3. Configurable class from mvc config (or a boolean on the config that forces a getBinding here w/o the try)?

Reply:
Also test?

Reply:
I don't know if it changes the performance profile or not. But I can see if I can find a way to inject a default impl.

Reply:
Made simpler, and will test.